scripts/initialise_demo.mjs

```diff
@@ -0,0 +1,289 @@
+    commitment_salt,
+    energy_consumed_wh,
+    production_end_time,
+    production_start_time,
+  }
+
+  const updateEndpoint = `http://localhost:${energyProviderPort}/v1/certificate/${id}`
+  const updateResult = await fetch(updateEndpoint, {
+    method: 'PUT',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(commitment),
+  })
+
+  if (!updateResult.ok) {
+    const message = `Error updating certificate ${id} on port ${energyProviderPort}. Error was ${updateResult.statusText}`
+    console.error(message)
+    throw new Error(message)
+  }
+
+  const issueEndpoint = `http://localhost:${energyProviderPort}/v1/certificate/${id}/issuance`
+  const issueResult = await fetch(issueEndpoint, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({}),
+  })
+  if (!issueResult.ok) {
+    const message = `Error issuing certificate ${id} on port ${energyProviderPort}. Error was ${issueResult.statusText}`
+    console.error(message)
+    throw new Error(message)
+  }
+
+  const waitForComplete = async () => {
+    await waitForCertificateState(id, 'issued', energyProviderPort)
+    const finalCert = await waitForCertificateState(
+      id,
+      'issued',
+      hydrogenProducerPort
+    )
+    return finalCert
+  }
+  return waitForComplete
+}
+
+const healthChecks = await Promise.all([
+  checkServiceHealth('hydrogen-producer-api', 8000),
+  checkServiceHealth('energy-owner-api', 8010),
+  checkServiceHealth('regulator-api', 8020),
+  checkServiceHealth('hydrogen-producer-identity', 9000),
+  checkServiceHealth('energy-owner-identity', 9010),
+  checkServiceHealth('regulator-identity', 9020),
+])
+if (!healthChecks.every((v) => v)) {
+  process.exit(1)
+}
+
+console.log('Fetching addresses')
+
+const [hydrogenSelf, energySelf, regulatorSelf] = await Promise.all([
+  getSelfIdentity(9000),
+  getSelfIdentity(9010),
+  getSelfIdentity(9020),
+])
+const identityMap = { hydrogenSelf, energySelf, regulatorSelf }
+
+console.log('Setting aliases')
+
+await Promise.all([
+  await setIdentities(identityMap, 9000),
+  await setIdentities(identityMap, 9010),
+  await setIdentities(identityMap, 9020),
+])
+
+console.log('Initiating certificates')
+
+const waitInitFns = []
+for (let i = 0; i < numberCertificatesToIssue; i++) {
+  const waitFn = await initiateCertificate(8000, 8010)
+  await delay(100)
+  waitInitFns.push(waitFn)
+}
+const certificates = await Promise.all(waitInitFns.map((f) => f()))
+
+console.log('Issuing certificates')
+
+const waitIssueFns = []
+for (const cert of certificates) {
+  const waitFn = await issueCertificate(cert, 8000, 8010)
+  await delay(100)
+  waitIssueFns.push(waitFn)
+}
+const finalCertificates = await Promise.all(waitIssueFns.map((f) => f()))
+
+for (const finalCert of finalCertificates) {
+  console.log(
+    `Created certificate ${finalCert.original_token_id} which is now in state ${finalCert.state}`
```

Comment:
This says `issued` in console, but when viewing the certificate list in the browser the state is `C. Embodiment`
<img width="1209" alt="Screenshot 2024-03-18 at 18 38 59" src="https://github.com/digicatapult/sqnc-hyproof-client/assets/39194639/b0f5efa2-62c3-4b86-a593-aa4f67871ac7">

Reply:
Fixed